### PR TITLE
Search Channel id by name

### LINKED ISSUE #1699: Slack Connector, search a channel ID, by its name
# Description
Currently, if we need to send any message to a Slack Channel, we have to know the channel ID. Unfortunately, this is not always possible, and it would be nice to have a way that Opsdroid translates channel names to channel ids.

I have been searching on the Slack API and there is no method that would do this, so the only thing I can think of, is iterating over all the channels. (probably when Opsdroid starts the connector?)  https://api.slack.com/methods/conversations.list
I would like to submit a PR for this, and I would love your thoughts, suggestions on how could this be better implemented.

## Steps to Reproduce
N/A

## Expected Functionality
```
await self.opsdroid.send(Message("Hello", target="test-channel"))
```

## Experienced Functionality
This will throw an error:
```
await self.opsdroid.send(Message("Hello", target="test-channel"))
```
Only works when channel ID is mentioned:
```
await self.opsdroid.send(Message("Hello", target="CF3ZP8KS8"))
```

## Versions
- **Opsdroid version:** All versions
- **Python version:** All Versions 
- **OS/Docker version:** All Versions

### PR DESCRIPTION
**Pull request by chillipeper**
# Description

Previously PR #1765  was opened for this, but it was never finished. This is the finished version of it.

Fixes #1699 

## Status
**READY** 

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Added relevant tests 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by FabioRosado — CHANGES_REQUESTED** [review]
Added some comments, let me know if you have bandwidth to address them, if not I'm happy to do a follow up PR 😄

File: docs/connectors/slack.md [hunk 1]
```diff
@@ -38,18 +39,33 @@ connectors:
...
+    # In order for bot-name and/or icon-emoji to work, the `chat:write.customize` 
+    # scope will have to be selected
+    bot-name: "mybot" # default "opsdroid" 
+    icon-emoji: ":smile:" # default ":robot_face:"
+
     default-room: "#random" # default "#general"
-    start-thread: false # default false. if true, opsdroid will start a thread when replying to a message
-```
 
-\* when `socket-mode` is true, you need to set also an `app-token` (more info: [app level tokens](https://api.slack.com/authentication/token-types#app))
```
[the remark this answers is not in the document]
Comment (chillipeper) [reply docs/connectors/slack.md:50]: this is not deleted, this was moved to line 46 and 47 of the modified version

File: docs/connectors/slack.md [hunk 2]
```diff
@@ -38,18 +39,33 @@ connectors:
...
+    # If set to true opsdroid will start a thread when replying to a message
+    start-thread: false # default false
 
-** In order for `bot-name` and/or `icon-emoji` to work, the `chat:write.customize` scope will have to be selected
```
[the remark this answers is not in the document]
Comment (chillipeper) [reply docs/connectors/slack.md:52]: same as above, lines 51-52

File: docs/connectors/slack.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (FabioRosado) [reply docs/connectors/slack.md:50]: Uh sorry! I'm trying the new view and I can't see that which is weird. Thank you for the patience and apologies!

**Review by FabioRosado — APPROVED** [review]
Thank you for working (or reviving this work and finishing it). Providing that I didn't bork the conflict resolution I will merge it once CI passes 😄

**Comment by FabioRosado** [comment]
Seems like I broke the tests, will fix them and push the changes if you are happy with it 👍

**Comment by FabioRosado** [comment]
Thank you again for implementing this!

**Pull request merged**